sa/storage-authority.go

```diff
@@ -232,6 +232,50 @@ func (ssa *SQLStorageAuthority) GetLatestValidAuthorization(registrationID int64
 	return ssa.GetAuthorization(auth.ID)
 }
 
+// GetValidAuthorizations returns the latest authorization object for all
+// domain names from the parameters that the account has authorizations for.
+func (ssa *SQLStorageAuthority) GetValidAuthorizations(registrationID int64, names []string, now time.Time) (latest map[string]core.Authorization, err error) {
+	if len(names) == 0 {
+		return nil, nil
+	}
+	params := make([]interface{}, len(names))
+	for i, name := range names {
+		id := core.AcmeIdentifier{Type: core.IdentifierDNS, Value: name}
+		tmp, err := json.Marshal(id)
+		if err != nil {
+			return nil, err
+		}
+		params[i] = string(tmp)
+	}
+
+	qmarks := strings.Repeat("?,", len(names))
```

Comment:
Similar to the other PR we just worked on, why not build an array of params and an array of question marks in parallel, then join the array of question marks? That makes it more obvious that there are the same number of each.